### PR TITLE
Added component support, and buying ammo

### PR DESCRIPTION
**Pull request by Sorata-senpai**
Now support components and ammo to be bought

### CODE REVIEW

File: client/main.lua [hunk 1]
```diff
@@ -62,57 +52,230 @@ function OpenBuyLicenseMenu(zone)
...
+	ShopOpen = true
+	ESX.UI.Menu.Open('default', GetCurrentResourceName(), 'gunshop_buy_weapons_components', {
+		title    = _U('gunshop_componenttitle'),
+		align    = 'top-left',
+		elements = buyAmmo
+	}, function(data, menu)
+		ESX.TriggerServerCallback('esx_weaponshop:buyWeapon', function(bought)
+			if bought then
+				if data.current.price > 0 then
+					--DisplayBoughtScaleform('ammo',nil, ESX.Math.GroupDigits(data.current.price))
```
Comment (ElPumpo) [inline client/main.lua]: What's this leftover?

File: client/main.lua [hunk 2]
```diff
@@ -62,57 +52,230 @@ function OpenBuyLicenseMenu(zone)
...
+	end
+
+
 	BeginScaleformMovieMethod(scaleform, 'SHOW_WEAPON_PURCHASED')
 
-	PushScaleformMovieMethodParameterString(_U('weapon_bought', ESX.Math.GroupDigits(price)))
-	PushScaleformMovieMethodParameterString(ESX.GetWeaponLabel(weaponName))
-	PushScaleformMovieMethodParameterInt(GetHashKey(weaponName))
+	PushScaleformMovieMethodParameterString(text)
+	if text2~= nil then
```
Comment (ElPumpo) [inline client/main.lua]: This looks awful tbh

File: config.lua [hunk 3]
```diff
@@ -30,10 +36,11 @@ Config.Zones = {
 
 	BlackWeashop = {
 		Legal = false,
-		Items = {},
+		Items = {
+			{ weapon = 'WEAPON_ADVANCEDRIFLE', components = { 0, 6000, 1000, 4000, 8000, nil }, price = 50000, ammoPrice = 20 }
```
Comment (ElPumpo) [inline config.lua]: Why not define the ammo to give when buying?

File: locales/fr.lua [hunk 4]
```diff
@@ -2,12 +2,22 @@ Locales ['fr'] = {
...
+  ['weapon_bought'] = 'Acheté pour %s EUR',
   ['not_enough_black'] = 'Vous n\'avez ~r~pas assez~s~ d\'argent sale',
   ['not_enough'] = 'vous n\'avez ~r~pas assez~s~ d\'argent',
-  ['already_owned'] = 'you already own this weapon!',
-  ['shop_menu_title'] = 'magasin',
+  ['already_owned'] = 'Vous possédez déjà cette arme!',
   ['shop_menu_prompt'] = 'appuyez sur ~INPUT_CONTEXT~ pour accéder au magasin.',
   ['shop_menu_item'] = '%s EUR',
   ['map_blip'] = 'armurerie',
+  ['buy_ammo'] = 'Acheter munitions:',
```
Comment (ElPumpo) [inline locales/fr.lua]: All translations begin lowercase you should know better

File: locales/pl.lua [hunk 5]
```diff
@@ -10,4 +10,13 @@ Locales ['pl'] = {
   ['shop_menu_prompt'] = 'Wciśnij ~INPUT_CONTEXT~ aby otworzyć sklep.',
   ['shop_menu_item'] = '%s PLN',
   ['map_blip'] = 'sklep z bronią',
+  ['gunshop_owned'] = 'owned',
+  ['gunshop_free'] = 'free',
+  ['gunshop_money'] = 'You don\'t have enough money',
+  ['gunshop_bought'] = 'you bought an ~y~%s~s~ for ~r~$%s~s~',
+  ['gunshop_item'] = '$%s',
+  ['gunshop_weapontitle'] = 'Weapon shop',
```
Comment (ElPumpo) [inline locales/pl.lua]: again lower case damnit

File: server/main.lua [hunk 6]
```diff
@@ -42,56 +17,108 @@ ESX.RegisterServerCallback('esx_weaponshop:buyLicense', function(source, cb)
...
+				end
+			else
+				print(('esx_weaponshop: %s attempted to buy an invalid weapon component.'):format(xPlayer.identifier))
+				cb(false)
+			end
 
+			--Weapon Ammo
+		elseif type == 3 then
+			if xPlayer.getMoney() >= selectedWeapon.ammoPrice then
+				xPlayer.removeMoney(selectedWeapon.ammoPrice)
```
Comment (ElPumpo) [inline server/main.lua]: Lol where's `xPlayer.addWeaponAmmo()`

File: server/main.lua [hunk 7]
```diff
@@ -42,56 +17,108 @@ ESX.RegisterServerCallback('esx_weaponshop:buyLicense', function(source, cb)
...
-	
 		end
-	end
-end)
+	else
+		-- Weapon
+		if type == 1 then
+			if xPlayer.getAccount('black_money').money >= selectedWeapon.price then
+				xPlayer.removeAccountMoney('black_money', selectedWeapon.price)
+				xPlayer.addWeapon(weaponName, 42)
```
Comment (ElPumpo) [inline server/main.lua]: No dont give any ammo.

File: server/main.lua (hunk 6 above)
Comment (KandaSoranyan) [reply server/main.lua]: wasn't it awful to do the function ? ahah

File: config.lua (hunk 3 above)
Comment (KandaSoranyan) [reply config.lua]: what do you mean ?

File: client/main.lua (hunk 1 above)
Comment (KandaSoranyan) [reply client/main.lua]: don't know wich one i should use for ammo

File: locales/fr.lua (hunk 4 above)
Comment (KandaSoranyan) [reply locales/fr.lua]: mistakes

File: client/main.lua (hunk 2 above)
Comment (KandaSoranyan) [reply client/main.lua]: not calling natives when not needed

File: server/main.lua (hunk 7 above)
Comment (KandaSoranyan) [reply server/main.lua]: Why ?

**Comment by DarkRainbowX** [comment]
Can you add the ability to buy ammo clip separately?

**Comment by LazarusOfTheFallen** [comment]
@Sorata-senpai If you are here, please resubmit this PR with updated code to resolve conflicts.

**Pull request closed without merging**